### PR TITLE
Integration tests: replace cb-helloblock with cb-insight and cb-blockr

### PR DESCRIPTION
**Pull request by weilu**
Blockr flips out on tx propagation sometimes, so using cb-insight for that. Why not use cb-insight for everything? Well, it doesn't support getting multiple transactions, yet. Feature wise, cb-blockr is still more complete. I'm also using a patched version of cb-insights, because latest version in npm does not 1) support transaction.get and 2) include unconfirmed amount in address.summary.

https://coconut-macaroon.herokuapp.com/ is just a faucet I spawn up with @jprichardson's neat bitcoin-faucet (patched as well :P)

### CODE REVIEW

**Comment by weilu** [comment]
This closes #399

File: test/integration/utils.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+function pollUnspent(blockchain, address, done) {
+  blockchain.addresses.unspents(address, function (err, unspents) {
+    if (err) return done(err)
+
+    if(unspents == null || unspents.length === 0) {
+      return setTimeout(function() {
+        pollUnspent(blockchain, address, done)
+      }, 200)
```
Comment (dcousens) [inline test/integration/utils.js]: Does this just go on forever?

**Comment by dcousens** [comment]
ACK, LGTM.

Unless @rubensayshi can give us a way to integrate with https://www.blocktrail.com/ that doesn't require an API token,  I'm happy with this.

@rubensayshi perhaps allow non-authenticated requests but just have them severely rate limited?

**Comment by jprichardson** [comment]
Looks good, other than @weilu butchering my precious faucet :p

**Comment by weilu** [comment]
@jprichardson FML haha your faucet needs more patching -- doesn't handle p2sh address man.

**Comment by dcousens** [comment]
Tests still failing?

**Comment by dcousens** [comment]
Tests pass.  Looks good.  Can we hold out on a response from @rubensayshi before merge?

**Comment by weilu** [comment]
sure.

**Comment by dcousens** [comment]
Merging, @rubensayshi still feel free to comment on the above.

**Pull request merged**